src/main/java/it/gov/pagopa/pu/classification/controller/AssessmentsController.java

```diff
@@ -1,26 +1,42 @@
 package it.gov.pagopa.pu.classification.controller;
 
 import it.gov.pagopa.pu.classification.controller.generated.AssessmentEntityExtendedControllerApi;
+import it.gov.pagopa.pu.classification.model.Assessments;
+import it.gov.pagopa.pu.classification.repository.AssessmentsRepository;
 import it.gov.pagopa.pu.classification.service.AssessmentsService;
 import it.gov.pagopa.pu.classification.util.SecurityUtils;
+import it.gov.pagopa.pu.debtposition.dto.generated.InstallmentNoPIIResponse;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.RestController;
 
-/** Controller to host spring-data-rest directly not supported methods */
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Controller to host spring-data-rest directly not supported methods
+ */
 @RestController
 public class AssessmentsController implements AssessmentEntityExtendedControllerApi {
 
   private final AssessmentsService assessmentsService;
+  private final AssessmentsRepository assessmentsRepository;
 
-  public AssessmentsController(AssessmentsService assessmentsService) {
+  public AssessmentsController(AssessmentsService assessmentsService, AssessmentsRepository assessmentsRepository) {
     this.assessmentsService = assessmentsService;
+    this.assessmentsRepository = assessmentsRepository;
   }
 
 
   @Override
   public ResponseEntity<Void> createAssessmentByReceiptId(Long receiptId) {
     String accessToken = SecurityUtils.getAccessToken();
-    assessmentsService.getInstallmentsByReceiptId(receiptId, accessToken);
+    List<InstallmentNoPIIResponse> installmentsList = assessmentsService.getInstallmentsByReceiptId(receiptId, accessToken);
+    List<Assessments> assessmentsList = new ArrayList<>();
+    installmentsList.forEach(installmentNoPIIResponse ->
+      assessmentsList.add(assessmentsService.getAssessment(installmentNoPIIResponse, accessToken))
+    );
```

Comment:
use stream API instead
```suggestion
    List<Assessments> assessmentsList = installmentsList.stream()
       .map(i ->
           assessmentsService.getAssessment(i, accessToken))
       .toList();
```